test/lib/pgBackRestTest/Module/Manifest/ManifestAllTest.pm

```diff
@@ -0,0 +1,1088 @@
+            MANIFEST_SUBKEY_CHECKSUM_PAGE, true);
+
+        # Default "master" is flipping because it's not something we read from disk
+        $oManifestExpected->boolSet(MANIFEST_SECTION_TARGET_FILE . ":default", MANIFEST_SUBKEY_MASTER, undef, false);
+        $oManifestExpected->boolSet(MANIFEST_SECTION_TARGET_FILE,  MANIFEST_TARGET_PGDATA . '/' . $strTest,
+            MANIFEST_SUBKEY_MASTER, true);
+
+        $oManifest->build(storageDb(), $self->{strDbPath}, $oLastManifest, true);
+        $self->testResult(sub {$self->manifestCompare($oManifestExpected, $oManifest)}, "", 'updates from last manifest');
+
+# CSHANG What is the [backup:option] option-checksum-page setting? Why are the checksum-page/error settings per file not dependent on this? What are these for?
+        # MANIFEST_SUBKEY_CHECKSUM_PAGE = false and MANIFEST_SUBKEY_CHECKSUM_PAGE_ERROR set/not set
+        #---------------------------------------------------------------------------------------------------------------------------
+        $oLastManifest->boolSet(MANIFEST_SECTION_TARGET_FILE,  MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_CHECKSUM_PAGE, false);
+
+        $oManifestExpected->boolSet(MANIFEST_SECTION_TARGET_FILE,  MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_CHECKSUM_PAGE, false);
+
+        $oManifest->build(storageDb(), $self->{strDbPath}, $oLastManifest, true);
+        $self->testResult(sub {$self->manifestCompare($oManifestExpected, $oManifest)}, "",
+            'checksum-page false, checksum-page-error not set');
+
+        my @iyChecksumPageError = (1);
+        $oLastManifest->set(MANIFEST_SECTION_TARGET_FILE,  MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_CHECKSUM_PAGE_ERROR, \@iyChecksumPageError);
+
+        $oManifestExpected->set(MANIFEST_SECTION_TARGET_FILE,  MANIFEST_TARGET_PGDATA . '/' . $strTestNew,
+            MANIFEST_SUBKEY_CHECKSUM_PAGE_ERROR, \@iyChecksumPageError);
+
+        $oManifest->build(storageDb(), $self->{strDbPath}, $oLastManifest, true);
+        $self->testResult(sub {$self->manifestCompare($oManifestExpected, $oManifest)}, "",
+            'checksum-page false, checksum-page-error set');
+    }
+
+    ################################################################################################################################
+    if ($self->begin('fileAdd()'))
+    {
+        my $oManifest = new pgBackRest::Manifest($strBackupManifestFile, {bLoad => false, strDbVersion => PG_VERSION_94});
+        my $oManifestExpected = dclone($oManifestBase);
+
+        $oManifest->build(storageDb(), $self->{strDbPath}, undef, true);
+
+        # Add a file after building manifest
+        my $lTimeTest = $lTime + 10;
+        storageDb()->put(storageDb()->openWrite($self->{strDbPath} . '/' . $strTest,
+            {strMode => MODE_0644, strUser => TEST_
```

Comment:
This info is not preserved during the archive process and is not very important in any case.  So - this is implemented as intended.

However, the way the mode is set is not very precise.  It would be better to mask rather than use a constant.  Added issue 3 for that.